cranelift/codegen/src/dominator_tree.rs

```diff
@@ -236,20 +250,40 @@ impl DominatorTree {
     /// Returns `true` if `block_a` dominates `block_b`.
     ///
     /// A block is considered to dominate itself.
-    fn block_dominates(&self, block_a: Block, mut block_b: Block) -> bool {
-        let pre_a = self.nodes[block_a].pre_number;
-
-        // Run a finger up the dominator tree from b until we see a.
-        // Do nothing if b is unreachable.
-        while pre_a < self.nodes[block_b].pre_number {
-            let idom = match self.idom(block_b) {
-                Some(idom) => idom,
-                None => return false, // a is unreachable, so we climbed past the entry
-            };
-            block_b = idom;
+    /// This uses preorder numbers for O(1) constant time performance.
+    pub fn block_dominates(&self, block_a: Block, block_b: Block) -> bool {
+        let na = &self.nodes[block_a];
+        let nb = &self.nodes[block_b];
+        na.dom_pre_number <= nb.dom_pre_number && na.dom_pre_max >= nb.dom_pre_max
+    }
+
+    /// Get an iterator over the direct children of `block` in the dominator tree.
+    ///
+    /// These are the blocks whose immediate dominator is `block`, ordered according
+    /// to the CFG reverse post-order.
+    pub fn children(&self, block: Block) -> ChildIter<'_> {
+        ChildIter {
+            domtree: self,
+            next: self.nodes[block].child,
         }
+    }
+}
+
+/// An iterator that enumerates the direct children of a block in the dominator tree.
```

Comment:
To avoid polluting the diff, could you keep this at the same point in the file? It looks like it's otherwise unmodified (modulo the type renaming).